### PR TITLE
Make new colorRange and colorDomain props experimental

### PR DESCRIPTION
**Pull request by 1chandu**
For #1566 

#### Background
- Make colorRange and colorDomain experimental
- Un deprecate minColor and maxColor.
- Remove un-used uniforms and corresponding unit-tests.

#### Change List
- Make new colorRange and colorDomain props experimental

### CODE REVIEW

File: docs/whats-new.md [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 Release date: TBD, target Q1-Q2, 2018
 
-### ScreenGridLayer
+### New coloring props for ScreenGridLayer **Experimental**
```
Comment (ibgreen) [inline docs/whats-new.md]: Nit: I'd keep the ScreenGridLayer heading, to keep the What's new organized.

The question is how much we want to talk about experimental features here. We generally avoid bringing attention to them but we can make exceptions where it makes sense.

File: test/src/core-layers/core-layers.spec.js [hunk 2]
```diff
@@ -66,23 +66,6 @@ test('ScreenGridLayer#constructor', t => {
             layer.state.cellScale === layer.state.model.uniforms.cellScale,
             'should update uniform cellScale'
           );
-          t.ok(
```
Comment (ibgreen) [inline test/src/core-layers/core-layers.spec.js]: Why did we need to cut a test?

File: docs/whats-new.md (hunk 1 above)
Comment (1chandu) [reply docs/whats-new.md]: I will update the heading,  but I do want to keep the section given enabling these experimental features change final output the layer rendering.

File: test/src/core-layers/core-layers.spec.js (hunk 2 above)
Comment (1chandu) [reply test/src/core-layers/core-layers.spec.js]: As part of supporting `colorRange` and `colorDomain`, color calculation is moved out of the shader, so I am removing these un-used uniforms and their update, hence remove these unit tests.

**Pull request merged**